WhisperAPI/WhisperAPI/Controllers/ContextController.cs

```diff
@@ -21,20 +21,29 @@ public ContextController(IContexts contexts)
 
         public override void OnActionExecuting(ActionExecutingContext actionExecutingContext)
         {
-            var searchQuery = (SearchQuery)actionExecutingContext.ActionArguments["searchQuery"];
+            Guid chatKey;
+            object param;
+            if (actionExecutingContext.ActionArguments.TryGetValue("searchQuery", out param))
+            {
+                var searchQuery = (SearchQuery) param;
+                log4net.ThreadContext.Properties["requestId"] = Guid.NewGuid();
+                if (!this.ModelState.IsValid)
```

Comment:
This should be checked at the beginning of the method, no matter what, we should do model validation upfront and then for the rest of the method we suppose model is validated.